rs/nns/governance/proto/ic_nns_governance/pb/v1/governance.proto

```diff
@@ -2086,6 +2088,41 @@ message StopOrStartCanister {
   optional CanisterAction action = 2;
 }
 
+message UpdateCanisterSettings {
+  // The target canister ID to call update_settings on. Required.
+  optional ic_base_types.pb.v1.PrincipalId canister_id = 1;
+
+  // Log visibility of a canister.
+  enum LogVisibility {
+    LOG_VISIBILITY_UNSPECIFIED = 0;
+    // The log is visible to the controllers of the dapp canister.
+    LOG_VISIBILITY_CONTROLLERS = 1;
+    // The log is visible to the public.
+    LOG_VISIBILITY_PUBLIC = 2;
+  }
```

Comment:
Is there an advantage to repeating the name of the type in the enum variant? Otherwise I'd do this:
```suggestion
  enum LogVisibility {
    UNSPECIFIED = 0;
    // The log is visible to the controllers of the dapp canister.
    CONTROLLERS = 1;
    // The log is visible to the public.
    PUBLIC = 2;
  }
```

Reply:
@anchpop I believe using this convention generates Rust as you'd expect, and that's why we have this pattern (i.e. the rust doesn't have the prefix)

Reply:
Actually, I think there is no reason to do it - I tried to generate rust without it, and they generate the same rust names and the only difference is the generated `from_str_name` and `as_str_name`.

I can only guess, but the reason might be this: https://protobuf.dev/programming-guides/style/#enums. It would be a perfectly good reason to do so for C++, since the generated C++ enum values won't contain the name of the enum. If we would stick to the pattern of providing `*_pb` canister methods, then we should also stick to this pattern since clients might use `.proto` files for generating their types. I think we are moving away from it but we are not quite there yet. I'd prefer to keep this pattern if it's just for the consistency (we have 18 enums here and all adhere to this pattern)